KFSPTS-24999 Add ObjectCode routing to doc types

cah292 opened the pull request:
This PR updates the appropriate doc type workflow XML files to include KualiCo's ObjectCode routing feature. Also, certain documents that don't necessarily have this routing (such as some of the PO variants) have still been updated to at least include the new app doc statuses related to this.

Note that these changes only affect the workflow XML files, which need to be manually ingested into KFS to have any effect. The changes do NOT affect runtime code, data dictionary configuration, OJB configuration, etc.

If the KualiCo patch gets merged prior to this PR, then I would like to rebase this feature branch afterwards.

cah292 commented:
NOTE: I've rebased the feature branch to bring it up-to-date with the 2022-10-19 patch changes. Also, as a result of one of the other code changes that went in prior to the patch, I ended up squash-merging the original changes onto a temporary branch, and then I force-pushed it to overwrite this branch. Everything should be correct now, but please let me know if you notice any corrupted changes as a result of this rework.

I still have the original changes available on the following branch:

https://github.com/CU-CommunityApps/cu-kfs/tree/KFSPTS-24999-backup

nkimble commented on src/main/resources/edu/cornell/kfs/module/purap/document/workflow/VendorCreditMemoDocument.xml:
```diff
@@ -40,7 +40,8 @@
                         <join name="JoinRequiresImageAttachment"/>
                     </split>
-                    <role name="Account" nextNode="TreasuryManager" nextAppDocStatus="Awaiting Treasury Manager Approval"/>
+                    <role name="Account" nextNode="ObjectCode"/>
```
Does this route node need a nextAppDocStatus attribute value to indicate it is "Awaiting Object Code Review"?

cah292 replied:
Vendor Credit Memo documents did not receive a new app doc status related to Object Code routing, so we don't need to add a "nextAppDocStatus" setting for it.

nkimble replied:
Thanks for clarifying.

nkimble reviewed (changes requested):
These changes look good. 

I had a couple questions regarding formatting and a possible missing status.

Also, there is a note in file [object code routing FINP's.png] attached to the main user story that states in order for this to work at Cornell, we need these XML changes ingested as well as having members added to the Object Code Reviewer Role per functional user request when they want to implement. I did not see a user story linked to this one for Nicole to setup that security. 
Does another user story need to be created for this security setup?
 **OR** 
Should there be sub-tasks for each environment added to this user story for that work?

cah292 commented on src/main/resources/edu/cornell/kfs/module/purap/document/workflow/PurchaseOrderAmendmentDocument.xml:
```diff
@@ -100,15 +102,17 @@
                         <join name="JoinRequiresContractManagementReview"/>
                     </split>
+                    <role name="ObjectCode" nextNode="RequiresBudgetReview"
+                          nextAppDocStatus="Awaiting Budget Approval"/>
```
I'll update the formatting for consistency.

cah292 commented on src/main/resources/edu/cornell/kfs/module/purap/document/workflow/PurchaseOrderDocument.xml:
```diff
@@ -85,15 +87,17 @@
                         <join name="JoinRequiresContractManagementReview"/>
                     </split>
+                    <role name="ObjectCode" nextNode="RequiresBudgetReview"
+                          nextAppDocStatus="Awaiting Budget Approval"/>
```
I'll update the formatting for consistency.

cah292 commented:
@nkimble With regards to the needed KIM adjustments for this, I had placed some instructions in the DEV/TEST validation sub-tasks, but I haven't created a sub-task or user story yet for the needed Production KIM changes. I'll go ahead and create an appropriate sub-task for the Production setup, which we can convert into a separate user story if necessary.

If you think the DEV/TEST setup should be in separate sub-tasks instead of being included in the validation sub-tasks, please let me know.

cah292 commented:
I believe I've implemented all of the needed changes. I also added a new sub-task to the user story for configuring the related role in Production. (If necessary, we can convert that sub-task into a distinct user story at a later time.)

This is ready for review again.
